### PR TITLE
Keep Yourself Alive! (on zookeeper)

### PR DESCRIPTION
**Pull request by stkem**

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala [hunk 1]
```diff
@@ -78,6 +83,23 @@ class ServiceDiscoveryImpl @Inject() (
...
+  private def stillRegistered(): Boolean = myNode.map{ node =>
+    clusters(services.currentService).instanceForNode(node).map(_ => true).getOrElse(false)
+  }.getOrElse(true)
+
+
+  private def keepAlive() : Unit = {
+    scheduler.scheduleOnce(2 minutes){
+      if (stillRegistered) {
+        keepAlive()
+      }
```
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: We usually write statements like `else`, `catch`, etc. on the same line as the closing brace so it's clear there's another case.

File: kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala [hunk 2]
```diff
@@ -78,6 +83,23 @@ class ServiceDiscoveryImpl @Inject() (
 
   override def myClusterSize: Int = clusters.get(services.currentService) map {c => c.size} getOrElse 0
 
+  private def stillRegistered(): Boolean = myNode.map{ node =>
```
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: I think something like this is more clear:

``` scala
myNode.flatMap { clusters(services.currentService).instanceForNode(_) }.isDefined
```

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: That will be false when myNode is None though, right? I want it to be true in that case.

File: kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: Good pattern. I'll change it here. Going to take some getting used to, but I like it.

File: kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala (hunk 2 above)
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: oh, you're right. This works though, I think.

``` scala
myNode map { clusters(services.currentService).instanceForNode(_).isDefined } getOrElse true
```

Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/zookeeper/ServiceDiscovery.scala]: I like it. Changed.

**Pull request merged**